### ISSUE TITLE
UnusedBlockArgument: harmful autocorrection. Disable autocorrection for SOME cops?

### ISSUE THREAD

**Issue by zverok** [rubocop/rubocop#11199]
Courtesy of production accident!

Imagine you have code like this:
```ruby
# id_groups is [1, 2, 3], [4, 5, 6], ...
id_groups.map { |ids| SomeModel.where(id: ids).order(...).select(...).to_a }
```
Then, imagine after some refactoring (scopes introduced and removed, code copy-pasted, whatever), we have instead, this:
```ruby
id_groups.map { |ids| SomeModel.order(...).select(...).to_a }
```
The problem is there is no more filtering by `ids`, we are selecting the whole table.
Rubocop is helpful here: it complains about unused block argument, and draws attention to a potential problem.

But what happens then: somebody finished a big chunk of code, made sure the specs are green, and then just `rubocop -A` whatever problems are there (because it is also stylistic/formatting problems). And Rubocop "helpfully" "solves" the unused block argument by renaming it to `_ids`.

Thinking a bit more about it, I tend to believe that auto-fixing "unused argument/variable" by renaming is **never** a good call: naming it `_ids` actually identifies "I (human developer) am conscious this argument wasn't used, and I declare that it is OK." So the automatic tool doing it seems like an oxymoron and defies the very meaning of the cop.

I would propose to remove autocorrection for this kind of cops whatsoever, but being aware that [it will break someone's workflow](https://xkcd.com/1172/), I am thinking that it might be possible to introduce some `Autocorrect: false` (or `NeverAutocorrect: true`) per-cop setting, so we could set it up in `.rubocop.yml`, overwriting whatever the cop believes useful to do?..

**Comment by dvandersluis** [comment]
You actually can already do this! Just set `AutoCorrect: false` for the cop.

```yaml
Style/Cop:
  AutoCorrect: false
```

You can also disable it for an entire department:

```yaml
Style:
  AutoCorrect: false
```

**Comment by dvandersluis** [comment]
This is also mentioned in the docs btw: https://docs.rubocop.org/rubocop/1.39/configuration.html#autocorrect

**Comment by zverok** [comment]
:facepalm: Ugh, sorry, I quick-googled that and somehow missed.
